File: ViewBase/Source/Validators/EmptyListValidator.vb
Imports FluentValidation

Public Class EmptyListValidator(Of T)
    Inherits AbstractValidator(Of IEnumerable(Of T))
    Sub New(propertyDisplayName As String)
        RuleFor(Function(list) list).
            Must(Function(list) list IsNot Nothing AndAlso list.Any).WithMessage($"{propertyDisplayName}: debe tener al menos un elemento.")
    End Sub
End Class

File: CoreView/src/Validators/EmptyListValidator.cs
using FluentValidation;
using System.Collections.Generic;
using System.Linq;

namespace Wells.CoreView.Validators
{
    public class EmptyListValidator<T> : AbstractValidator<IEnumerable<T>>
    {
        public EmptyListValidator(string propertyDisplayName)
        {
            RuleFor((list) => list).Must((list) => list != null && list.Any()).WithMessage($"{propertyDisplayName}: debe tener al menos un elemento.");
        }
    }
}
